apps/docs/contributing/services/agent/architecture.mdx

```diff
@@ -0,0 +1,14 @@
+---
+title: Architecture
+---
+
+The agent is deployed as a cluster of machines to fly.io. Each machine is identical and capable of serving any incoming request.
+
+We run a gossip network between all machines for service discovery and failure detection.
+The gossip network will also be used for broadcasting configuration changes and cache invalidations.
+
+On top of the gossip layer, sits a consistent hash ring for sharding any kind of data across the cluster.
+
+<Warning>
+TODO: andreas - add more details about the architecture
```

Comment:
Can we change this Warning to something that isn't a todo for yourself. 

"Details of architecture coming soon"